FileShareHelper.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.File;
+using NLog.Common;
+using System;
+using System.Threading.Tasks;
+
+namespace NLog.Extensions.AzureFileSystem
+{
+    public static class FileShareHelper
+    {
+        public static CloudFileShare InitializeFileShare(IAzureFileSystemTarget target, CloudFileShare fileShare, string folderName, string fileName)
+        {
+            return AsyncHelper.RunSync(async () => await InitializeFileShareAsync(target, fileShare, folderName, fileName));
+        }
+
+        public static async Task<CloudFileShare> InitializeFileShareAsync(IAzureFileSystemTarget target, CloudFileShare fileShare, string folderName, string fileName)
+        {
+            if (fileShare == null)
+            {
+                fileShare = await CreateStorageAccountFromConnectionStringAsync(target)
+                    .ConfigureAwait(false);
+            }
+
+            folderName = folderName.Replace(@"\", @"/");
+
+            var rootDir = fileShare.GetRootDirectoryReference();
+
+            // Get a reference to the directory.
+            var folder = rootDir.GetDirectoryReference(folderName);
+            var folderExists = await folder.ExistsAsync().ConfigureAwait(false);
+
+            if (!folderExists)
+            {
+                var levels = folderName.Split(new char[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+                var currentLevel = string.Empty;
+
+                for (int i = 0; i < levels.Length; i++)
+                {
+                    if (currentLevel == string.Empty) currentLevel = levels[i];
```

Comment:
no need for this line if currentLevel is initialised

Reply:
now I just check the i variable